test/utilities/fillIn.test.ts

```diff
@@ -0,0 +1,28 @@
+import { ReactNode } from 'react';
+
+import { RuleObject } from 'rc-field-form/lib/interface';
+
+import { fillInFieldConfig } from '../../src/utilities';
+import { IFieldConfigPartial } from '../../src/interfaces';
+
+function getRequiredMessage(fieldConfig: IFieldConfigPartial): string | ReactNode {
+  const requiredFieldConfig = fillInFieldConfig(fieldConfig),
+    validator: RuleObject = requiredFieldConfig.formValidationRules.required as RuleObject;
+
+  return validator.message;
+}
+
+describe('fillIn', () => {
+  it('Correctly handles required value', async () => {
+    const required = true,
+      field = 'field';
+
+    expect(getRequiredMessage({ required, field })).toBe(`'Field' is required`);
+
+    expect(getRequiredMessage({ required, field, label: undefined })).toBe(`'Field' is required`);
+
+    expect(getRequiredMessage({ required, field, label: 'Law Firm' })).toBe(`'Law Firm' is required`);
+
+    expect(getRequiredMessage({ required, field, label: 'Law Firm', showLabel: false })).toBe(`'Law Firm' is required`);
```

Comment:
How should `showLabel` affect the resulting message? I don't see it being used.

Reply:
It's not. I guess this is just a check against future stuff that might mess with things. Or to at least acknowledge this use case.